icicle/appUtils/ntt/ntt.cu

```diff
@@ -514,6 +534,7 @@ namespace ntt {
         reverse_input = true;
         break;
       case Ordering::kNR:
+      case Ordering::kNM:
```

Comment:
Don't see `kMN` among the cases. Also kind of weird that ordering is done here for radix 2 but one function deeper for mixed radix, maybe we can do some unification here? (speaking of, `dit` in mixed radix version is the same as `ct_butterfly` in radix2, just difference in lingo)

Reply:
kMN is not there for the exact same reason kRN isn't. The code was like that so I did not modify it but basically the default values of ct_butterfly and reverse_input are for those cases already. Probably cleaner to add those cases and not rely on the default values. I will do that.

About consolidation, there is asymmetry between the two algs where radix-2 is implemented in ntt.cu while mixed-radix is not. Ideally they are each in a different file and may share common logic but I avoided refactoring too much to avoid rebase problems for other PRs (like the multi-gpu that is also modifying ntt.cu).

Regarding this switch case specifically, I can try to consolidate but I am afraid it would cause more confusion than good and that's why I did not. Note that calculating the input/output/coset reorder and dit/dif bit and passing it to mixed-radix NTT seems worse to me compared to passing the ordering field and let the specific implementation decide how it emulates R etc. Also note that except for dit (which is ct_buttefrly ineed), radix2 doesn't care about output/coset reorder the way mixed-radix does and even input reorder is not always the same (mainly R emulation requires reorder for mixed-radix but not for radix2).
If you think it's important I will try to see how it can be cleanly done but my feeling is that this is not a good idea unless refactoring this entire kind of "strategy-pattern" more.

Reply:
I did not consolidate the switch cases since I think it will do more bad than good but I did split the radix-2 implementation so it's more symmetric now.

Reply:
Definitely understand not wanting to refactor the new NTT and agree with _passing the ordering field and let the specific implementation decide how it emulates R etc_. Maybe then we can move reversals inside and rename ct_butterfly to dit in the old NTT?